doc/source/jupyterhub/customizing/user-environment.md

```diff
@@ -69,28 +69,157 @@ If you'd like users to select an environment from **multiple docker images**,
 see {ref}`multiple-profiles`.
 ```
 
+(user-interfaces)=
+
+## Selecting a user interface
+
+[JupyterLab][] is the new user interface for Jupyter,
+which is meant to replace the classic notebook user interface (UI).
+While users already can interchange `/tree` and `/lab` in the URL to switch between
+the classic UI and JupyterLab if both are installed,
+Deployments using JupyterHub 1.x and earlier default to the classic UI,
+while 2.0 will make JupyterLab the default.
+
+[jupyterlab]: https://jupyterlab.readthedocs.io
+
+There are two things to customize, when picking the user interface to launch for users:
+
+1. the user interface (UI)
+2. the server program to launch
+
+There are two main Jupyter server implementations.
+_Most_ deployments will not see a difference,
+but there can be issues for certain server extensions.
```

Comment:
Do you know of a good external page we could link to that explains the potential problems?

Reply:
I don't know of good docs covering what is likely to break, other than possible use of private APIs or assumptions not preserved across the transition. Most things are meant to work, and [this doc][] describes migrating an extension. I don't actually know of any examples of classic server extensions that don't work today with the new server. Maybe I should be softer and say something more like:

> Jupyter Server aims to be as backward-compatible as possible,
> so the vast majority of server extensions continue to work across the upgrade.
> If you experience issues, however, please report them to the Jupyter Server team,
> and you can use the classic server in the meantime.


[this doc]: https://jupyter-server.readthedocs.io/en/latest/developers/extensions.html#migrating-an-extension-to-use-jupyter-server

Reply:
```suggestion
but there can be issues for certain server extensions. If unsure, new applications should choose `jupyter_server`._):
```